src/mongoose_hooks.erl

```diff
@@ -368,8 +368,11 @@ register_user(HostType, LServer, LUser) ->
     LUser :: jid:luser(),
     Result :: mongoose_acc:t().
 remove_user(Acc, LServer, LUser) ->
+    Params = #{user => LUser, server => LServer},
+    Args = [LUser, LServer],
+    ParamsWithLegacyArgs = ejabberd_hooks:add_args(Params, Args),
     HostType = mongoose_acc:host_type(Acc),
-    run_hook_for_host_type(remove_user, HostType, Acc, [LUser, LServer]).
+    run_hook_for_host_type(remove_user, HostType, Acc, ParamsWithLegacyArgs).
```

Comment:
All right, this is a hook that I always wanted to touch and never could, so this is a good chance 😃 
The issue is that here we pass user and server, not knowing if they have been normalised already. See how so many subscribers will do again and again `jid:nameprep` and `jid:nodeprep`, or `jid:make` with both, just so that we can be sure they were normalised.

I think that the legacy arguments should stay as they are, but, the new argument should simply be a jid, a-la `Jid = jid:make_bare(User, Server)`, and that way subscribers can be sure the string has been normalised.

In the case of the handler for ejabberd_sm it might feel redundant, we'll need to pattern match on the jid `#jid{luser = LUser, lserver = LServer}` to extract them, but at this point we can be sure we could omit nameprepping them later on.